### PR TITLE
feat($rootScope): Add $once method

### PR DESCRIPTION
**Pull request by noam3127**
This listens on events just like $on, but will deregister
the listener automatically upon the first call.

### CODE REVIEW

**Comment by noam3127** [comment]
I signed it!

File: src/ng/rootScope.js [hunk 1]
```diff
@@ -1125,14 +1125,58 @@ function $RootScopeProvider() {
...
+       *   - `stopPropagation` - `{function=}`: calling `stopPropagation` function will cancel
+       *     further event propagation (available only for events that were `$emit`-ed).
+       *   - `preventDefault` - `{function}`: calling `preventDefault` sets `defaultPrevented` flag
+       *     to true.
+       *   - `defaultPrevented` - `{boolean}`: true if `preventDefault` was called.
+       *
+       * @param {string} name Event name to listen on.
+       * @param {function(event, ...args)} listener Function to call when the event is emitted.
+       */
+      $once: function(name, listener) {
```
Comment (Narretz) [inline src/ng/rootScope.js]: To keep it dry, `$once` should share most of its code with `$on`.

**Comment by Narretz** [comment]
It's essentially sugar, so I am not a huge fan of adding it. Let's see what others say.

**Comment by lgalfaso** [comment]
I think this can be added as a third party module. With this in mind, I agree with @Narretz that it should not be added

File: src/ng/rootScope.js (hunk 1 above)
Comment (noam3127) [inline src/ng/rootScope.js]: Updated the code to make it dry, @Narretz

File: src/ng/rootScope.js [hunk 2]
```diff
@@ -1109,30 +1109,50 @@ function $RootScopeProvider() {
...
+        var self = this;
+        var modifiedListener = function() {
+          deregisterListener(self, name, modifiedListener);
+          listener.apply(self, arguments);
```
Comment (lgalfaso) [inline src/ng/rootScope.js]: why use `self` when calling the listener?

File: src/ng/rootScope.js (hunk 1 above)
Comment (lgalfaso) [inline src/ng/rootScope.js]: I think that what @Narretz is trying to say is that `$once` should be implemented as

```js
// Untested code
$once: function(name listener) {
  var delegate = this.$on(name, function() {
    delegate();
    listener.apply(null, arguments);
  });
},
```

and leave `$on` unmodified.

**Comment by lgalfaso** [comment]
I still do not like the idea of adding the `$once` method. It should be possible to add this method thru a third party module by modifying the prototype.

**Comment by Narretz** [comment]
Ok, let's close this. This is good for a 3rd party module.

**Pull request closed without merging**